.github/scripts/trymerge.py

```diff
@@ -1109,6 +1109,35 @@ def __init__(self, message: str, rule: Optional['MergeRule'] = None) -> None:
         super().__init__(message)
         self.rule = rule
 
+class MergeError(Exception):
```

Comment:
Hmm, it's not really a `MergeError`, but rather `ExceptionWithReason`? And shouldn't it be subclassed from `RuntimeError` rather than `Exception`?

Reply:
The error ends up capturing things which are not RuntimeErrors (as it just replaces `handleException`) like MandatoryChecksMissingError, so probably best to subclass to Exception. Though I'll change the name to make it more clear.